src/main/java/GenericsAndCllections/User.java

```diff
@@ -0,0 +1,31 @@
+package GenericsAndCllections;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+public class User {
+    public long userId;
+    public String username;
+    public Map<String, Address> addresses;
+
+    public User(String username, Map<String, Address> addresses) {
+        Random rand = new Random();
+        this.userId = rand.nextInt(999);
+        this.username = username;
+        this.addresses = new HashMap<String, Address>();
```

Comment:
in this case the new HashMap will create an empty map, but I see that you have an addresses parameter to this constructor. I suggest to set the addresses to this.addresses instead of the HashMap initialization.

Reply:
done